File: Ex18_SQlite01_10130492/app/src/main/java/com/example/ex18_sqlite01_10130492/SQLiteHelper.java
package com.example.ex18_sqlite01_10130492;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

public class SQLiteHelper extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "comments2.db"; //資料庫一定要.db
    private static final int DATABASE_VERSION = 0;
    public static final String COLUMN_ID = "_id";
    public static final String TABLE_COMMENTS = "comments";
    public static final String COLUMN_COMMENTS = "comment";

    private static final String DATABASE_CREATE = String.format("CREATE TABLE %s (%s INTEGER PRIMARY KEY AUTOINCREMENT, %s TEXT NOT NULL);", TABLE_COMMENTS,COLUMN_ID,COLUMN_COMMENTS);
    public SQLiteHelper(@Nullable Context context, @Nullable SQLiteDatabase.CursorFactory factory) {
        super(context, DATABASE_NAME, factory, DATABASE_VERSION);

    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(DATABASE_CREATE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        onCreate(db);
    }
}

File: AppQuiz_10130492_20221228_p4_final/app/src/main/java/com/example/appquiz_10130492_20221228/databases/SQLiteHelper.kt
package com.example.appquiz_10130492_20221228.databases

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

class SQLiteHelper(context: Context?): SQLiteOpenHelper(context,DATABASE_NAME, null, DATABASE_VERSION ) {
    val assignTableName = "Assignments"
    private val SQL_CREATE_ENTRIES_COURSES = "CREATE TABLE $assignTableName (name TEXT,"+
            " cityName TEXT, "+
            " tel TEXT, "+
            " address TEXT, "+
            "description TEXT)"
    private val SQL_DELETE_ENTRIES_ASSIGN = "DROP TABLE IF EXISTS $assignTableName"
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(SQL_CREATE_ENTRIES_COURSES)
        Log.d("DB","Creating Databases")
    }
    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // This database is only a cache for online data, so its upgrade policy is
        // to simply to discard the data and start over
        db.execSQL(SQL_DELETE_ENTRIES_ASSIGN)
        onCreate(db)
    }
    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }
    companion object {
        // If you change the database schema, you must increment the database version.
        const val DATABASE_VERSION = 2
        const val DATABASE_NAME = "School.db"

    }
}